syntax = "proto3";

package hex_ai.proto;

enum PieceType {
  kEmpty = 0;
  kHorizontal = 1;
  kVertical = 2;
}

message HexBoard {
  // Size of one side of the board
  int32 size = 1;

  // Should be exactly size * size
  repeated PieceType positions = 3;
}

message HexState {
  HexBoard board = 1;
  // Should not be kEmpty.
  PieceType to_move = 2;
}
